client/webserver/site/src/js/markets.ts

```diff
@@ -1008,6 +1021,68 @@ export default class MarketsPage extends BasePage {
     } else this.page.submitBttn.textContent = intl.prep(intl.ID_SET_BUTTON_BUY, { asset: Doc.shortSymbol(this.market.baseCfg.unitInfo.conventional.unit) })
   }
 
+  setOrderBttnEnabled (isEnabled: boolean, disabledTooltipMsg?: string) {
+    const btn = this.page.submitBttn
+    if (isEnabled) {
+      btn.removeAttribute('disabled')
+      btn.removeAttribute('title')
+    } else {
+      btn.setAttribute('disabled', 'disabled')
+      if (disabledTooltipMsg) btn.setAttribute('title', disabledTooltipMsg)
+    }
+  }
+
+  updateOrderBttnState () {
+    const { market: mkt, currentOrder: { qty: orderQty, rate: orderRate, isLimit, sell } } = this
+    const baseWallet = app().assets[this.market.base.id].wallet
+    const quoteWallet = app().assets[mkt.quote.id].wallet
+    if (!baseWallet || !quoteWallet) return
+
+    if (orderQty <= 0) {
+      this.setOrderBttnEnabled(false, intl.prep(intl.ID_ORDER_BUTTON_QTY_ERROR))
+      return
+    }
+
+    // Market orders
+    if (!isLimit) {
+      if (sell) {
+        this.setOrderBttnEnabled(orderQty <= baseWallet.balance.available, intl.prep(intl.ID_ORDER_BUTTON_SELL_BALANCE_ERROR))
+      } else {
+        this.setOrderBttnEnabled(orderQty <= quoteWallet.balance.available, intl.prep(intl.ID_ORDER_BUTTON_BUY_BALANCE_ERROR))
+      }
+      return
+    }
+
+    if (!orderRate) {
+      this.setOrderBttnEnabled(false, intl.prep(intl.ID_ORDER_BUTTON_QTY_RATE_ERROR))
+      return
+    }
+
+    // Limit sell
+    if (sell) {
+      if (baseWallet.balance.available < mkt.cfg.lotsize) {
+        this.setOrderBttnEnabled(false, intl.prep(intl.ID_ORDER_BUTTON_SELL_BALANCE_ERROR))
+        return
+      }
+      if (mkt.maxSell) {
+        this.setOrderBttnEnabled(orderQty <= mkt.maxSell.swap.value, intl.prep(intl.ID_ORDER_BUTTON_SELL_BALANCE_ERROR))
+      }
+      return
```

Comment:
Seems like we're just giving up here if we don't have a `mkt.maxSell`. Is that OK? I guess we're just expecting the `preSell` to come back soon if we don't already have it?

Reply:
Yes, once `/api/maxsell` comes back the previous branch will execute.